tmt/steps/report/reportportal.py

```diff
@@ -1,152 +1,221 @@
-        except (tmt.utils.GeneralError, KeyError):
-            message = response.text
-        if not response.ok:
-            raise tmt.utils.ReportError(
-                f"Received non-ok status code from ReportPortal, response text is: {message}")
 
-        self.debug(f"Response code from the server: {response.status_code}")
-        self.debug(f"Message from the server: {message}")
-        self.info("report", "Successfully uploaded.", "yellow")
+            # Create a launch
+            self.info("launch", launch_name, color="cyan")
+            response = session.post(url=f"{url}/launch",
+                                    headers=headers,
+                                    json={"name": launch_name,
+                                          "description": self.step.plan.summary,
+                                          "attributes": attributes,
+                                          "startTime": launch_time})
+            self.handle_response(response)
+            launch_uuid = yaml_to_dict(response.text).get("id")
+            assert launch_uuid is not None
+            self.info("uuid", launch_uuid, "yellow", shift=1)
+
+            # For each test
+            for result in self.step.plan.execute.results():
+                test = [test for test in self.step.plan.discover.tests()
+                        if test.serialnumber == result.serialnumber][0]
+                # TODO: for happz, connect Test to Result if possible
+
+                item_attributes = attributes.copy()
+                if test.contact:
+                    item_attributes.append({"key": "contact", "value": test.contact[0]})
+                env_vars = [{'key': k, 'value': v} for k, v in test.environment.items()
+                            if not k.startswith("TMT_") or tmt_vars]
+
+                #  Create a test item
+                self.info("test", result.name, color="cyan")
+                response = session.post(url=f"{url}/item",
+                                        headers=headers,
+                                        json={"name": result.name,
+                                              "description": test.summary,
+                                              "attributes": item_attributes,
+                                              "parameters": env_vars,
+                                              "codeRef": test.web_link() or None,
+                                              "launchUuid": launch_uuid,
+                                              "type": "step",
+                                              "testCaseId": test.id or None,
+                                              "startTime": result.starttime})
+                self.handle_response(response)
+                item_uuid = yaml_to_dict(response.text).get("id")
+                assert item_uuid is not None
+                self.info("uuid", item_uuid, "yellow", shift=1)
```

Comment:
Is the `uuid` information that much interesting? What about moving the message to `self.verbose()`?

Reply:
You're right, `uuid` is only for testing purposes and for future follow-up providing an access to it via API, `verbose()` should be perfectly fine for it.